src/test/java/GuiAppTest/ActionSetterTest.java

```diff
@@ -0,0 +1,33 @@
+package GuiAppTest;
+
+import CommandLineApp.NumberConverter;
+import GuiApp.ActionSetter;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ActionSetterTest {
+
+    @Test
+    public void setsActionOnButton() {
+        ConvertButtonDouble buttonDouble = new ConvertButtonDouble();
+        ActionSetter actionSetter = new ActionSetter(new NumberConverter(), new TextFieldDouble(), new AppLabelDouble());
+
+        actionSetter.addClickHandler(buttonDouble);
+
+        assertTrue(buttonDouble.wasCalled);
+    }
+
+    @Test
+    public void updatesLabelWhenButtonIsPressed() {
+        ConvertButtonDouble buttonDouble = new ConvertButtonDouble();
+        AppLabelDouble label = new AppLabelDouble();
+        ActionSetter actionSetter = new ActionSetter(new NumberConverter(), new TextFieldDouble(), label);
+
+        actionSetter.addClickHandler(buttonDouble);
+        buttonDouble.press();
+
+        assertEquals("XLIV", label.getText());
```

Comment:
Where does XLIV come from? The test would feel more complete if the connection between "You entered the number 44", "you press the button", and "You see XLIV" was more direct.
Currently it feels like the punchline to a great joke, but the setup is missing 😸

Reply:
Ok, I get what you mean... it'll take longer to figure out how to change this though :)